setup.sh

```diff
@@ -0,0 +1,74 @@
+#!/bin/bash
+time=`TZ=Asia/Jerusalem date '+%Y%m%d_%H%M'`
+VAG_COMMON_DIR="/vagrant"
+HOME="/home/vagrant"
+LOG_DIR="$VAG_COMMON_DIR/log"
+OUTFILE="$LOG_DIR/${time}_vagrant-out.log"
+FLASK_APP_PY="$VAG_COMMON_DIR/flask_init.py"
+FLASK_DIR="$HOME/FlaskApp"
+FLASK_EXEC="$HOME/runFlask.sh"
+KILLFLASK_SCRIPT="$VAG_COMMON_DIR/killFlask.sh"
+PYTHON3="`which python3`"
+FLASK_ONBOOT=false # Currently Flask On Startup feature disabled [ Not Supported ]
+
+runSetup() {
+    step=1
+
+    if [ ! -e $FLASK_APP_PY ]; then
+        echo "ERROR: Missing $FLASK_APP_PY file..." >> $OUTFILE
+        exit 1
+    else
+        chmod +x $FLASK_APP_PY
+    fi
+
+    if [ ! -d $FLASK_DIR ]; then
+        mkdir $FLASK_DIR
+    fi
+
+    if [ ! -d $LOG_DIR ]; then
+        mkdir $LOG_DIR
+    else
+        rm -f $LOG_DIR/*vagrant-out.log
+    fi
+    touch $OUTFILE
+    # Adding execution permission to $KILLFLASK_SCRIPT 
+    if [ -e $KILLFLASK_SCRIPT ]; then
+        chmod +x $KILLFLASK_SCRIPT
+    fi
+
+    echo " "
+    echo " "
+    echo "==> Running Provision Script: $0:"
+    echo " "
+    echo "$((step++)). Installing & Activating Python-3 Virtual Environment..."
+    sudo apt-get install -yq python3-venv >> $OUTFILE 2>&1 
```

Comment:
why sudo is required?

Reply:
In case it would require to write system directories.
In this instance I admit I did not try it without sudo.
I will test it and omit it if redundant.

Reply:
Indeed found sudo to be mandatory for all required apt-get and pip operations for permissions purposes of certain paths.

Reply:
It is mandatory when you provision with ":privileged => false".  Remove it and you would find out that sudo is not required.